sentry-android-core/src/test/java/io/sentry/android/core/ScreenshotEventProcessorTest.kt

```diff
@@ -200,5 +202,81 @@ class ScreenshotEventProcessorTest {
         assertFalse(fixture.options.sdkVersion!!.integrationSet.contains("Screenshot"))
     }
 
+    @Test
+    fun `when screenshots are captured rapidly, capturing should be debounced`() {
+        CurrentActivityHolder.getInstance().setActivity(fixture.activity)
+
+        val processor = fixture.getSut(true)
+        val event = SentryEvent().apply {
+            exceptions = listOf(SentryException())
+        }
+        val hint0 = Hint()
+        processor.process(event, hint0)
+        assertNotNull(hint0.screenshot)
+
+        val hint1 = Hint()
+        processor.process(event, hint1)
+        assertNull(hint1.screenshot)
+    }
+
+    @Test
+    fun `when screenshots are captured rapidly, debounce flag should be propagated`() {
+        CurrentActivityHolder.getInstance().setActivity(fixture.activity)
+
+        var debounceFlag = false
+        fixture.options.setBeforeScreenshotCaptureCallback { _, _, debounce ->
+            debounceFlag = debounce
+            true
+        }
+
+        val processor = fixture.getSut(true)
+        val event = SentryEvent().apply {
+            exceptions = listOf(SentryException())
+        }
+        val hint0 = Hint()
+        processor.process(event, hint0)
+        assertFalse(debounceFlag)
+
+        val hint1 = Hint()
+        processor.process(event, hint1)
+        assertTrue(debounceFlag)
+    }
+
+    @Test
+    fun `when capture callback returns false, no screenshot should be captured`() {
+        CurrentActivityHolder.getInstance().setActivity(fixture.activity)
+
+        fixture.options.setBeforeScreenshotCaptureCallback { _, _, _ ->
+            false
+        }
+        val processor = fixture.getSut(true)
+
+        val event = SentryEvent().apply {
+            exceptions = listOf(SentryException())
+        }
+        val hint = Hint()
+
+        processor.process(event, hint)
+        assertNull(hint.screenshot)
+    }
+
+    @Test
+    fun `when capture callback returns true, a screenshot should be captured`() {
+        CurrentActivityHolder.getInstance().setActivity(fixture.activity)
+
+        fixture.options.setBeforeViewHierarchyCaptureCallback { _, _, _ ->
+            true
+        }
+        val processor = fixture.getSut(true)
+
+        val event = SentryEvent().apply {
+            exceptions = listOf(SentryException())
+        }
+        val hint = Hint()
+
+        processor.process(event, hint)
+        assertNotNull(hint.screenshot)
+    }
+
```

Comment:
could you add a last test `when capture callback returns true, debounce flag is overwritten`?
basically a test where the debounce flag is false and the callback returns true, so the view screenshot should be attached anyway